scripts/fetch/fetch.precip_spatial.R

```diff
@@ -0,0 +1,64 @@
+fetch.precipSpatial <- function(viz = as.viz('precip-spatial')){
+
+  view <- readDepends(viz)[["view-limits"]]
+  
+  ### constants
+  cell_size <- .1 # degrees -- set to make grid more course or dense.
+  states <- c("florida", "georgia", "alabama", "south carolina") # from maps states id list.
+  ### /constants
+    
+  bbox <- view[["latlonbbox"]]
+  
+  x_range <- (bbox[3] - bbox[1])/cell_size
+  y_range <- (bbox[4] - bbox[2])/cell_size
+  
+  grid_topology <- sp::GridTopology(bbox[c(1:2)],
+                                    cellsize = c(cell_size,cell_size),
+                                    cells.dim = c(x_range, y_range))
+  sp_grid <- raster::raster(sp::SpatialGrid(grid_topology, sp::CRS("+init=epsg:4326")))
+  
+  sp_cells <- raster::rasterToPolygons(sp_grid)
+  sp_points <- sp::SpatialPoints(raster::rasterToPoints(sp_grid),sp::CRS("+init=epsg:4326"))
+  
+  state_boundary <- to_sp("state", proj.string = "+init=EPSG:4326")
+  
+  state_IDs <- sapply(slot(state_boundary, "polygons"), function(x) slot(x, "ID"))
+  index <- which(state_IDs %in% states)
+  index <- c()
+  for(i in 1:length(state_boundary@polygons)) {
+    if(state_boundary@polygons[[i]]@ID %in% states) {
+      index <- c(index,i)
+    }
+  }
+  state_boundary <- SpatialPolygonsDataFrame(state_boundary[index,], data.frame(viz=rep("viz", length(index)), 
+                                                                                stringsAsFactors = F, row.names = states))
+  state_boundary <- maptools::unionSpatialPolygons(state_boundary, state_boundary@data$viz)
+  
+  sp_cells@proj4string <- CRS("+init=epsg:4326")
```

Comment:
can you use the shared EPSG code in `view-limits`?

Reply:
Figured I'd do everything in here as 4326 then project later. But happy to do whatever works.

Reply:
I'm switching over the the projection from the project.